code/game/objects/items/stacks/stack.dm

```diff
@@ -8,35 +8,81 @@
 /*
  * Stacks
  */
-
 /obj/item/stack
+	icon = 'icons/obj/stack_objects.dmi'
 	gender = PLURAL
-	origin_tech = "materials=1"
 	var/list/datum/stack_recipe/recipes
 	var/singular_name
 	var/amount = 1
-	var/max_amount //also see stack recipes initialisation, param "max_res_amount" must be equal to this max_amount
-	var/stack_id //used to determine if two stacks are of the same kind.
+	var/max_amount = 50 //also see stack recipes initialisation, param "max_res_amount" must be equal to this max_amount
+	var/merge_type // This path and its children should merge with this stack, defaults to src.type
+	var/number_of_extra_variants = 0 //Determines whether the item should update it's sprites based on amount.
+	origin_tech = "materials=1"
+
+
+/obj/item/stack/New(loc, new_amount)
```

Comment:
Why not Initialize?

Reply:
Can't simply substitute it without substituting a lot of others, due to `..()` calls.
It's something I'd like to do for all objects, but it would produce a very unatomized PR if done here.